### PR TITLE
WIP: [primtorch] Adding multigammaln ref

### PR DESCRIPTION
**Pull request by zaxtax**
This adds the `multigammaln` op as a ref. This is congruence with what jax does.

### CODE REVIEW

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -646,6 +646,14 @@ def logsumexp(
     return result
 
 
+@register_decomposition(torch.ops.aten.special_multigammaln)
+@out_wrapper()
+def multigammaln(a: TensorLikeType, dim: int) -> TensorLikeType:
+    c = 0.25 * dim * (dim - 1) * prims.log(torch.pi)
+    b = prims.arange(0, dim) / 2
+    return sum(prims.lgamma(a - b), -1) + c
```
Comment (kshitij12345) [inline torch/_refs/__init__.py:654]: I think the implementation should be moved to https://github.com/pytorch/pytorch/blob/master/torch/_refs/special/__init__.py

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -16296,6 +16296,11 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
                    ),
                    # lgamma have multiple singularities at x <= 0
                    reference_numerics_filter=NumericsFilter(condition=lambda x: x < 0.1, safe_val=1)),
+    OpInfo('multigammaln',
+           dtypes=floating_and_complex_types(),
+           supports_out=False,
+           supports_forward_ad=True,
+           supports_fwgrad_bwgrad=True),
```
Comment (kshitij12345) [inline torch/testing/_internal/common_methods_invocations.py:16303]: There are existing OpInfos for `mvlgamma/multigammaln` (So probably, a new entry shouldn't be required)

https://github.com/pytorch/pytorch/blob/ca9919e3e81bb563422beb83e61b71cb8deca62c/torch/testing/_internal/common_methods_invocations.py#L13367-L13386

File: torch/_refs/__init__.py (hunk 1 above)
Comment (Lezcano) [reply torch/_refs/__init__.py:654]: also, call `torch.lgamma`, `torch.log`, etc. These call will then be transformed into refs / prims as needed by a context manager.

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (zaxtax) [reply torch/testing/_internal/common_methods_invocations.py:16303]: I was getting a no OpInfo found error. Maybe I need to update how I use PythonRefsInfo

**Comment by Lezcano** [comment]
Superseded by https://github.com/pytorch/pytorch/pull/85153

**Pull request closed without merging**